src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/CloudStackCloudUtils.java

```diff
@@ -49,61 +49,77 @@ public static String doRequest(@NotNull CloudStackHttpClient cloudStackHttpClien
-                    throw new FogbowException();
-                }
+
+        int countTries = 0;
+        CloudStackQueryAsyncJobResponse response = getAsyncJobResponse(
+                client, cloudStackUrl, jobId, cloudStackUser);
+        while(response.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
+            if (countTries >= MAX_TRIES) {
+                throw new TimeoutCloudstackAsync(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
             }
-            throw new TimeoutCloudstackAsync(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
-        } else {
-            throw new UnexpectedException();
+            sleepThread();
+            response = getAsyncJobResponse(client, cloudStackUrl, jobId, cloudStackUser);
+            countTries++;
         }
+
+        return processJobResult(response, jobId);
     }
 
-    public static String processJobResult(CloudStackQueryAsyncJobResponse queryAsyncJobResult,
-                                      String jobId)
+    @VisibleForTesting
+    static String processJobResult(@NotNull CloudStackQueryAsyncJobResponse response,
+                                   String jobId)
             throws FogbowException {
-        switch (queryAsyncJobResult.getJobStatus()){
+
+        switch (response.getJobStatus()){
             case CloudStackQueryJobResult.SUCCESS:
-                return queryAsyncJobResult.getJobInstanceId();
+                return response.getJobInstanceId();
             case CloudStackQueryJobResult.FAILURE:
                 throw new FogbowException(String.format(Messages.Exception.JOB_HAS_FAILED, jobId));
             default:
                 throw new UnexpectedException(Messages.Error.UNEXPECTED_JOB_STATUS);
         }
     }
 
-    public static CloudStackQueryAsyncJobResponse getAsyncJobResponse(CloudStackHttpClient client,
+    @VisibleForTesting
+    public static CloudStackQueryAsyncJobResponse getAsyncJobResponse(@NotNull CloudStackHttpClient client,
                                                                       String cloudStackUrl,
                                                                       String jobId,
-                                                                      CloudStackUser cloudStackUser)
+                                                                      @NotNull CloudStackUser cloudStackUser)
             throws FogbowException {
 
         String jsonResponse = CloudStackQueryJobResult.getQueryJobResult(
                 client, cloudStackUrl, jobId, cloudStackUser);
         return CloudStackQueryAsyncJobResponse.fromJson(jsonResponse);
     }
 
+    @VisibleForTesting
+    protected static long getTimeSleepThread() {
+        return ONE_SECOND_IN_MILIS;
+    }
+
+    private static void sleepThread() {
+        try {
+            Thread.sleep(getTimeSleepThread());
+        } catch (InterruptedException e) {
+            LOGGER.warn("Thread waiting was interrupeted", e);
```

Comment:
Maybe "Thread's not able to sleep" is better

Reply:
Ok. I will improve this.
I will make this change in the last PR(five), right?

Reply:
Done